### PR TITLE
[#32002] Update "Social icon" module

### PR DESCRIPTION
**Pull request by kimpepper**
Social links should contain correct classes:

```html
<div class="item-list">
<ul class="social__list">
<li class="social__item first"><span class="fa fa-facebook"></span><a href="http://www.facebook.com/pages/PreviousNext/170975076260877">Facebook</a></li>
<li class="social__item"><span class="fa fa-twitter"></span><a href="http://www.twitter.com/PreviousNext">Twitter</a></li>
<li class="social__item"><span class="fa fa-youtube"></span><a href="http://www.youtube.com">Youtube</a></li>
<li class="social__item"><span class="fa fa-vimeo-square"></span><a href="http://www.vimeo.com">Vimeo</a></li>
<li class="social__item"><span class="fa fa-flickr"></span><a href="http://www.flickr.com">Flickr</a></li>
<li class="social__item"><span class="fa fa-rss"></span><a href="/rss.xml">RSS Feed</a></li>
<li class="social__item last"><span class="fa fa-envelope"></span><a href="/contact">Email</a></li>
</ul></div>
```

### CODE REVIEW

File: modules/custom/agov_social_links/agov_social_links.module [hunk 1]
```diff
@@ -56,6 +56,23 @@ function agov_social_links_services_list() {
...
+  $social_font = array(
+    'facebook' => 'fa-facebook',
+    'twitter' => 'fa-twitter',
+    'youtube' => 'fa-youtube',
+    'vimeo' => 'fa-vimeo-square',
+    'flickr' => 'fa-flickr',
+    'rss' => 'fa-rss',
+    'email' => 'fa-envelope',
+  );
+
```
Comment (kimpepper) [inline modules/custom/agov_social_links/agov_social_links.module]: Add a hook_alter() here to allow additional social icon support.

File: modules/custom/agov_social_links/agov_social_links.module [hunk 2]
```diff
@@ -143,18 +161,31 @@ function agov_social_links_block_view($delta = '') {
...
+          $items[$service] = array(
+            'data' => drupal_render($service_link),
+            'class' => array('social__item'),
+          );
         }
       }
 
+      if (!empty($items)) {
+        $block['content'] = theme('item_list', array(
```
Comment (kimpepper) [inline modules/custom/agov_social_links/agov_social_links.module]: Avoid calling theme(), use a nested render array instead.

**Comment by kimpepper** [comment]
Would be nice to convert theme functions to templates if we are making changes here.

File: modules/custom/agov_social_links/agov_social_links.module [hunk 3]
```diff
@@ -143,18 +161,31 @@ function agov_social_links_block_view($delta = '') {
...
+          $icon = agov_social_links_get_fontawesome($service);
+          $service_link = array(
             '#theme' => 'agov_social_link',
             '#title' => $service_title,
             '#icon' => $icon,
             '#url' => $service_url,
           );
+
+          $items[$service] = array(
+            'data' => drupal_render($service_link),
```
Comment (kimpepper) [inline modules/custom/agov_social_links/agov_social_links.module]: Do we need to call drupal_render() here? Can we just return a nested render array?

**Comment by kimpepper** [comment]
Would be good to add cache permanent to the block info as well.

**Comment by chuyenlv** [comment]
thanks @kimpepper, I will update

**Pull request merged**